test/foundry/Deposit.t.sol

```diff
@@ -0,0 +1,431 @@
+        for (uint256 i = 0; i < logs.length; i++) {
+            if (logs[i].topics[0] == DepositWithReceipt.selector) {
+                // Decode the event data
+                (
+                    address sender,
+                    address owner,
+                    uint256 assets,
+                    uint256 shares,
+                    uint256 id,
+                    bytes memory receiptInformation
+                ) = abi.decode(logs[i].data, (address, address, uint256, uint256, uint256, bytes));
+                eventFound = true;
+                eventData = DepositWithReceiptEvent({
+                    sender: sender,
+                    owner: owner,
+                    assets: assets,
+                    shares: shares,
+                    id: id,
+                    receiptInformation: receiptInformation
+                });
+                break;
+            }
+        }
+        uint256 expectedShares = aliceAssets.fixedPointMul(shareRatio, Math.Rounding.Up);
+
+        // Assert that the event log was found
+        assertTrue(eventFound, "DepositWithReceipt event log not found");
+
+        assertEqUint(vault.totalSupply(), vault.totalAssets());
+        assertEq(eventData.sender, alice);
+        assertEq(eventData.owner, alice);
+        assertEq(eventData.assets, aliceAssets);
+        assertEq(eventData.shares, expectedShares);
+        assertEq(eventData.id, 1);
+        assertEq(eventData.receiptInformation, fuzzedReceiptInformation);
+
+        vm.stopPrank();
+    }
+
+    function testMinShareRatio(
+        uint256 fuzzedKeyAlice,
+        string memory assetName,
+        string memory assetSymbol,
+        uint256 aliceAssets,
+        bytes memory receiptInformation
+    ) external {
+        // Ensure the fuzzed key is within the valid range for secp256k1
+        fuzzedKeyAlice = bound(fuzzedKeyAlice, 1, SECP256K1_ORDER - 1);
+        address alice = vm.addr(fuzzedKeyAlice);
+
+        uint256 shareRatio = 1e18;
+
+        // Prank as Alice for the transaction
+        vm.startPrank(alice);
+
+        // Assume that aliceAssets is not 0
+        vm.assume(aliceAssets != 0);
+
+        OffchainAssetReceiptVault vault = OffchainAssetVaultCreator.createVault(factory, alice, assetName, assetSymbol);
+
+        vault.grantRole(vault.DEPOSITOR(), alice);
+        vm.expectRevert(abi.encodeWithSelector(MinShareRatio.selector, shareRatio + 1, shareRatio));
+        vault.deposit(aliceAssets, alice, shareRatio + 1, receiptInformation);
+
+        vm.stopPrank();
+    }
+
+    function testZeroAssetsAmount(
+        uint256 fuzzedKeyAlice,
+        string memory assetName,
+        string memory assetSymbol,
+        bytes memory receiptInformation
+    ) external {
+        // Ensure the fuzzed key is within the valid range for secp256k1
+        fuzzedKeyAlice = bound(fuzzedKeyAlice, 1, SECP256K1_ORDER - 1);
+        address alice = vm.addr(fuzzedKeyAlice);
+        uint256 aliceAssets = 0;
```

Comment:
fuzz this?

Reply:
I'm testing testZeroAssetsAmount here, thats why aliceAssets is set to 0